lib/checkunusedvar.cpp

```diff
@@ -1622,10 +1623,11 @@ bool CheckUnusedVar::isFunctionWithoutSideEffects(const Function& func, const To
         // check nested function
         const Function* bodyFunction = bodyToken->function();
         if (bodyFunction) {
-            if (bodyFunction->fullName() == func.fullName()) { // recursion found
+            if (std::find(nested_in.begin(), nested_in.end(), bodyFunction) != nested_in.end()) { // recursion found
                 continue;
             }
-            if (!isFunctionWithoutSideEffects(*bodyFunction, *bodyToken)) {
+            nested_in.push_back(bodyFunction);
```

Comment:
I think you need to refactoring this somehow.. as far as I see it will think there is recursion here:

    void foo() {
        bar();
        bar();
    }

Reply:
As I see, this is not an issue.
If `bar()` is side-effect function, check will `return` at first entry.
If `bar()` is function without side-effects, second check will return same non-side-effect result as a first one and it's fine to skip it.

But `nested_in` variable can be renamed to something like `checked_funcs` to better display above-mentioned behavior.
What do you think about it?

Reply:
yes true. Yes I think `checkedFuncs` would be a better name.